### PR TITLE
[loxam_fr] Created a sitemapspider for Loxam in France (1049 items)

### PR DESCRIPTION
**Pull request by ckpiper**
<details><summary>Stats</summary>

```python
{'atp/brand/Loxam': 1049,
 'atp/brand_wikidata/Q3264407': 1049,
 'atp/category/shop/tool_hire': 1049,
 'atp/field/email/missing': 1049,
 'atp/field/image/missing': 1049,
 'atp/field/opening_hours/missing': 1,
 'atp/field/operator/missing': 1049,
 'atp/field/operator_wikidata/missing': 1049,
 'atp/field/state/missing': 1049,
 'atp/field/twitter/missing': 1049,
 'atp/nsi/perfect_match': 1049,
 'downloader/request_bytes': 527869,
 'downloader/request_count': 1053,
 'downloader/request_method_count/GET': 1053,
 'downloader/response_bytes': 24230468,
 'downloader/response_count': 1053,
 'downloader/response_status_count/200': 1053,
 'elapsed_time_seconds': 1275.127449,
 'finish_reason': 'finished',
 'finish_time': datetime.datetime(2024, 1, 18, 19, 15, 24, 242285, tzinfo=datetime.timezone.utc),
 'httpcompression/response_bytes': 148254088,
 'httpcompression/response_count': 1053,
 'item_scraped_count': 1049,
 'log_count/INFO': 30,
 'memusage/max': 278564864,
 'memusage/startup': 137310208,
 'request_depth_max': 1,
 'response_received_count': 1053,
 'robotstxt/request_count': 1,
 'robotstxt/response_count': 1,
 'robotstxt/response_status_count/200': 1,
 'scheduler/dequeued': 1052,
 'scheduler/dequeued/memory': 1052,
 'scheduler/enqueued': 1052,
 'scheduler/enqueued/memory': 1052,
 'start_time': datetime.datetime(2024, 1, 18, 18, 54, 9, 114836, tzinfo=datetime.timezone.utc)}
```
</details>

### CODE REVIEW

**Review by Cj-Malone — CHANGES_REQUESTED** [review]
Please can you take a look at StructuredDataSpider. Also probably don't need to list all the site maps. There is usually a mail one, or robots.txt

**Comment by ckpiper** [comment]
Got the images by using structured data spider.

<details><summary>New Stats</summary>

```python
{'atp/brand/Loxam': 1049,
 'atp/brand_wikidata/Q3264407': 1049,
 'atp/category/shop/tool_hire': 1049,
 'atp/field/email/missing': 1049,
 'atp/field/image/dropped': 4,
 'atp/field/image/missing': 4,
 'atp/field/opening_hours/missing': 1,
 'atp/field/operator/missing': 1049,
 'atp/field/operator_wikidata/missing': 1049,
 'atp/field/state/missing': 1049,
 'atp/field/twitter/missing': 1049,
 'atp/nsi/perfect_match': 1049,
 'downloader/request_bytes': 527869,
 'downloader/request_count': 1053,
 'downloader/request_method_count/GET': 1053,
 'downloader/response_bytes': 24229864,
 'downloader/response_count': 1053,
 'downloader/response_status_count/200': 1053,
 'elapsed_time_seconds': 1277.545061,
 'finish_reason': 'finished',
 'finish_time': datetime.datetime(2024, 1, 18, 20, 20, 46, 276551, tzinfo=datetime.timezone.utc),
 'httpcompression/response_bytes': 148252295,
 'httpcompression/response_count': 1053,
 'item_scraped_count': 1049,
 'log_count/DEBUG': 2113,
 'log_count/INFO': 30,
 'memusage/max': 280117248,
 'memusage/startup': 137519104,
 'request_depth_max': 1,
 'response_received_count': 1053,
 'robotstxt/request_count': 1,
 'robotstxt/response_count': 1,
 'robotstxt/response_status_count/200': 1,
 'scheduler/dequeued': 1052,
 'scheduler/dequeued/memory': 1052,
 'scheduler/enqueued': 1052,
 'scheduler/enqueued/memory': 1052,
 'start_time': datetime.datetime(2024, 1, 18, 19, 59, 28, 731490, tzinfo=datetime.timezone.utc)}
```
</details>

**Comment by ckpiper** [comment]
There is a robots.txt, but it sends you to the sitemap.xml which then sends you to the locationsitemap.xml

**Comment by davidhicks** [comment]
In a `post_process_item` override method, could you add `item.pop("image", None)` to get rid of the image?

It appears these are random stock art pictures or unrelated to the location, examples:

https://cdn.thumbor.leadformance.com/media/clients/5a338dffff4a420011ad4126/cda7b9b6-baf1-422c-9896-e33d64b250a0-0b99825a-ded6-4779-a730-aeee83517fc6-1496933473-5706.JPG  (used multiple times for different locations)

https://cdn.thumbor.leadformance.com/media/clients/5a338dffff4a420011ad4126/c56df71e-99ba-4ff0-98b6-ea5d3f353759-2021-03-02.jpg  (not the same brand?)


Only one location image I tried looked possibly legit:

https://cdn.thumbor.leadformance.com/media/clients/5a338dffff4a420011ad4126/3ceb6046-a27c-4d82-bc7e-738e9047547d-20170602-160635.jpg  (actual location of https://agence.loxam.fr/625-loxam-vichy maybe?)

File: locations/spiders/loxam_fr.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+class LoxamFrSpider(SitemapSpider, StructuredDataSpider):
+    name = "loxam_fr"
+    item_attributes = {"brand": "Loxam", "brand_wikidata": "Q3264407"}
+    allowed_domains = ["agence.loxam.fr"]
+    sitemap_urls = [
+        "https://agence.loxam.fr/locationsitemap1.xml",
+        "https://agence.loxam.fr/locationsitemap2.xml",
+        "https://agence.loxam.fr/locationsitemap3.xml",
+    ]
```
Comment (Cj-Malone) [inline locations/spiders/loxam_fr.py:14]:
```suggestion
    sitemap_urls = ["https://agence.loxam.fr/robots.txt"]
    sitemap_follow = ["locationsitemap"]
```
This way we are safe when `locationsitemap4.xml` gets added.

Comment (ckpiper) [reply locations/spiders/loxam_fr.py]: Oh. I did not know sitemap_follow was an option. Not many spiders have it. Thanks for the feedback!

**Review by Cj-Malone — APPROVED** [review]
Code looks fine. Thanks.

**Pull request merged**